product_company_default/models/product_template.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    company_id = fields.Many2one(default=lambda self: self._get_default_company())
+
+    @api.model
+    def _get_default_company(self):
```

Comment:
I guess this is non-blocking but according to the convention: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#models
```suggestion
    def _default_company_id(self):
```